src/main/java/dev/mccue/magicbean/processor/AnnotationProcessor.java

```diff
@@ -72,39 +72,67 @@ public boolean equals(Object o) {
                                         
                         """.formatted(
                         className,
-                        fields.stream()
-                                .map(field -> "java.util.Objects.equals(%s.%s, other.%s)".formatted(
-                                        selfExpr,
-                                        field.getSimpleName(),
-                                        field.getSimpleName()
-                                ))
-                                .collect(Collectors.joining(" && \n                   "))
+                        createEqualsExpressions(selfExpr, fields)
                 )
         );
         equalsAndHashCodeMethods.append("""
                             @Override
                             public int hashCode() {
-                                return java.util.Objects.hash(
-                                  %s
-                                );
+                                %s
                             }
                         
                         """.formatted(
-                        fields.stream()
-                                .map(field -> "      " + selfExpr + "." + field.getSimpleName())
-                                .collect(Collectors.joining(",\n          "))
+                        createHashMethodBody(selfExpr, fields)
                 )
         );
         return equalsAndHashCodeMethods.toString();
     }
 
+    private String createEqualsExpressions(String selfExpr, List<VariableElement> fields) {
+        if (fields.isEmpty())
+            return "super.equals(o)";
+
+        return fields.stream()
+                .map(field -> "java.util.Objects.equals(%s.%s, other.%s)".formatted(
+                        selfExpr,
+                        field.getSimpleName(),
+                        field.getSimpleName()
+                ))
+                .collect(Collectors.joining(" && \n                   "));
+    }
```

Comment:
We don't have a formal style guide, but can we structure this as 

```java
if (fields.isEmpty()) {
    ...
}
else {
    ...
}
```

Reply:
Done